target/linux/ramips/dts/mt7628an_hilink_hlk-7688a-gateway.dts

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "hilink,hlk-7688a-gateway", "hilink,hlk-7688a", "mediatek,mt7628an-soc";
+	model = "Hi-Link HLK-7688A IoT Gateway";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x8000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	buttonkeys {
+		compatible = "gpio-keys";
+
+		wdt_rst {
+			label = "wdt_rst";
+			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	aliases {
+		led-boot = &wlan;
+		led-failsafe = &wlan;
+		led-running = &wlan;
+		led-upgrade = &wlan;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wlan: status {
+			label = "hlk-7688a:green:wlan";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&state_default {
+	gpio {
+		groups = "i2c";
+		function = "gpio";
+	};
+};
+
+&uart1 {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```

Comment:
Can this go faster?

Reply:
U-Boot runs the SPI clock at 32 MHz.  Practical limit seems to be around 66 MHz.